### PR TITLE
Refactor(ci): Speed up pipeline and make them more readable using emojis

### PR DESCRIPTION
**Pull request by literat**
## Description

* it also includes a better caching of node_modules
* all jobs are now done in under 4 minutes
* emojis are used for better readability
* some jobs were separated to run in parallel

> [!IMPORTANT]
> Please, check the job names to see if they make sense to you.

### Additional context

- https://www.linkedin.com/pulse/continuous-integration-pipelines-frontend-monorepos-opi-danihelka-svbke/?trackingId=Z91USzlXTDCEpbI0LAVZ1A%3D%3D
- https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action

### CODE REVIEW

**Comment by literat** [comment]
> Beautiful! 👏🏻 I love the abstraction of actions, thanks for inspiration 🙂.

Thanks. I have also applied some newly gained knowledge about caching dependencies. I have summarized it in this post - https://literat.dev/blog/2024-09-13/how-to-cache-not-just-yarn-installs-on-github-actions/

**Review by curdaj — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ env:
 
 jobs:
   haslabel:
-    name: analyse labels
+    name: 🏷️ Labels Analyzing
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/e2e.yml:19]: Fixed.

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -33,6 +33,7 @@ jobs:
           contains: 'run-visual-tests'
 
   test:
+    name: 🧪 End-to-End Testing
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/e2e.yml:36]: Good point. Changed.

File: .github/workflows/git.yaml [hunk 3]
```diff
@@ -1,9 +1,10 @@
-name: Git Check
+name: 🔍 Git Checks
 
 on: [pull_request]
 
 jobs:
   block-fixup-merge:
+    name: 🚫 Blocked Merge by Fixup
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/git.yaml:7]: Fixed.

File: .github/workflows/test-web-twig.yaml [hunk 4]
```diff
@@ -16,22 +16,14 @@ on:
 
 jobs:
   prepare-icons:
+    name: 📱 Icons Preparing
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/test-web-twig.yaml:19]: Well, naming is hard. Now is the space to suggest better naming but I want to be consistent across all the jobs.

File: .github/workflows/test-web-twig.yaml [hunk 5]
```diff
@@ -95,7 +87,7 @@ jobs:
           ~/.composer/vendor/bin/php-coveralls --coverage_clover=coverage/clover.xml --json_path=coverage/coveralls-upload.json -v
 
   finalize-tests:
-    name: Tests finished
+    name: 🏁 Test Finalizing
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/test-web-twig.yaml:90]: Fixed.

**Review by pavelklibani — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -45,6 +46,7 @@
     "packages:lint": "lerna run lint --parallel --no-sort",
     "packages:start": "lerna run start --parallel",
     "packages:test": "lerna run test",
+    "packages:test:unit": "lerna run test:unit --ignore '@lmc-eu/spirit-web-twig'",
```
Comment (crishpeen) [inline package.json:49]: why is it ignored?

**Review by crishpeen — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (literat) [reply package.json:49]: Good point. Because of the usage in GitHub Actions workflow. But I have done this better. Fixed.

**Pull request merged**